src/views/CountryRiskWatch/RiskBarChart/CombinedChart/index.tsx

```diff
@@ -0,0 +1,384 @@
+    );
+
+    const selectedRiskData = useMemo(
+        () => {
+            if (selectedRiskMetricDetail.key === 'displacement') {
+                return listToMap(
+                    riskData?.idmc
+                        ?.map(getDataWithTruthyHazardType)
+                        ?.filter(isDefined) ?? [],
+                    (data) => data.hazard_type,
+                );
+            }
+
+            if (selectedRiskMetricDetail.key === 'riskScore') {
+                return {
+                    ...listToMap(
+                        riskData?.inform_seasonal
+                            ?.map(getDataWithTruthyHazardType)
+                            ?.filter(isDefined)
+                            ?.map((riskItem) => ({
+                                ...riskItem,
+                                ...mapToMap(
+                                    monthNumberToNameMap,
+                                    (_, monthName) => monthName,
+                                    (monthName) => riskScoreToCategory(
+                                        riskItem?.[monthName],
+                                        riskItem?.hazard_type,
+                                    ),
+                                ),
+                            })) ?? [],
+                        (data) => data.hazard_type,
+                    ),
+                    WF: {
+                        ...wfRiskDataItem,
+                        ...mapToMap(
+                            monthNumberToNameMap,
+                            (_, monthName) => monthName,
+                            (monthName) => riskScoreToCategory(
+                                wfRiskDataItem?.[monthName],
+                                'WF',
+                            ),
+                        ),
+                    },
+                };
+            }
+
+            const rasterDisplacementData = listToMap(
+                riskData?.raster_displacement_data
+                    ?.map(getDataWithTruthyHazardType)
+                    ?.filter(isDefined) ?? [],
+                (datum) => datum.hazard_type,
+            );
+
+            if (isNotDefined(fiRiskDataItem)) {
+                return rasterDisplacementData;
+            }
+
+            return {
+                ...rasterDisplacementData,
+                FI: fiRiskDataItem,
+            };
+        },
+        [riskData, selectedRiskMetricDetail, fiRiskDataItem, wfRiskDataItem],
+    );
+
+    const filteredRiskData = useMemo(
+        () => {
+            if (isNotDefined(selectedHazardType)) {
+                return selectedRiskData;
+            }
+
+            const riskDataItem = selectedRiskData[selectedHazardType];
+
+            return {
+                [selectedHazardType]: riskDataItem,
+            };
+        },
+        [selectedRiskData, selectedHazardType],
+    );
+
+    const maxValue = useMemo(
+        () => {
+            const maxValueList = Object.values(filteredRiskData ?? {}).map(
```

Comment:
Let's handle chase when filteredRiskData is not defined or empty.
Math.max will return Infinity in that case.